File: modules/spring-core/testdata/java/inspection/eventListener/src/com/event/listener/TestEventListenerWithRecursive.java
package com.event.listener;

import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Component
public class TestEventListenerWithRecursive {

	@EventListener
	public @interface CustomEventListener {
	}

	@CustomEventListener
	void test0() {
	}

	@CustomEventListener
	protected void test1() {
	}

	@CustomEventListener
	private void test2() {
	}

	@CustomEventListener
	public void test3() {
	}

	@CustomEventListener
	public void test4(String param1, String param2) {
	}

	@CustomEventListener
	private void test5(String param1, String param2) {
	}

	@CustomEventListener
	public void test6(String param1) {
	}
}

File: modules/spring-core/testdata/kotlin/inspection/eventListener/src/com/event/listener/TestEventListenerWithRecursive.kt
package com.event.listener

import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class TestEventListenerWithRecursive {
    @EventListener
    annotation class CustomEventListener

    @CustomEventListener
    fun test0() {
    }

    @CustomEventListener
    protected fun test1() {
    }

    @CustomEventListener
    private fun test2() {
    }

    @CustomEventListener
    fun test3() {
    }

    @CustomEventListener
    fun test4(param1: String?, param2: String?) {
    }

    @CustomEventListener
    private fun test5(param1: String, param2: String) {
    }

    @CustomEventListener
    fun test6(param1: String?) {
    }
}